models/repo/pin.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2016 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"encoding/json"
+
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/log"
+)
+
+//Pin Repo or unpin repo
+func PinRepo(ownerID, repoID int64, pin bool) error {
+
+	log.Info("pinning %v %v %v", ownerID, repoID, pin)
+	var newpinned []int64
+	if pin {
+		newpinned = append(newpinned, repoID)
+	}
+	pinnedstring, err := user_model.GetUserSetting(ownerID, "pinned")
```

Comment:
1. Add that constant to `models/user/setting_keys.go`.
1. Rename it to `pinned_repos`.